src/scripting/natives/natives_entities.cpp

```diff
@@ -256,6 +278,272 @@ void AddEntityIOEvent(ScriptContext& script_context)
+    const char* key = script_context.GetArgument<const char*>(1);
+    KeyValuesType_t type = script_context.GetArgument<KeyValuesType_t>(2);
+
+    switch (type) {
+        case counterstrikesharp::TYPE_BOOL:
+        {
+            script_context.SetResult(keyValues->GetBool(key));
+        } break;
+
+        case counterstrikesharp::TYPE_INT:
+        {
+            script_context.SetResult(keyValues->GetInt(key));
+        } break;
+
+        case counterstrikesharp::TYPE_UINT:
+        {
+            script_context.SetResult(keyValues->GetUint(key));
+        } break;
+
+        case counterstrikesharp::TYPE_INT64:
+        {
+            script_context.SetResult(keyValues->GetInt64(key));
+        } break;
+
+        case counterstrikesharp::TYPE_UINT64:
+        {
+            script_context.SetResult(keyValues->GetUint64(key));
+        } break;
+
+        case counterstrikesharp::TYPE_FLOAT:
+        {
+            script_context.SetResult(keyValues->GetFloat(key));
+        } break;
+
+        case counterstrikesharp::TYPE_DOUBLE:
+        {
+            script_context.SetResult(keyValues->GetDouble(key));
+        } break;
+
+        case counterstrikesharp::TYPE_STRING:
+        {
+            script_context.SetResult(keyValues->GetString(key));
+        } break;
+
+        case counterstrikesharp::TYPE_POINTER:
+        {
+            script_context.SetResult(keyValues->GetPtr(key));
+        } break;
+
+        case counterstrikesharp::TYPE_STRING_TOKEN:
+        {
+            script_context.SetResult(keyValues->GetStringToken(key).GetHashCode());
+        } break;
+
+        case counterstrikesharp::TYPE_EHANDLE:
+        {
+            script_context.SetResult(keyValues->GetEHandle(key));
+        } break;
+
+        case counterstrikesharp::TYPE_COLOR:
+        {
+            script_context.SetResult(new Color(keyValues->GetColor(key)));
+            break;
+        }
+
+        case counterstrikesharp::TYPE_VECTOR:
+        {
+            script_context.SetResult(new Vector(keyValues->GetVector(key)));
+            break;
+        }
+
+        case counterstrikesharp::TYPE_VECTOR2D:
+        {
+            script_context.SetResult(new Vector2D(keyValues->GetVector2D(key)));
+            break;
+        }
+
+        case counterstrikesharp::TYPE_VECTOR4D:
+        {
+            script_context.SetResult(new Vector4D(keyValues->GetVector4D(key)));
+            break;
+        }
+
+        case counterstrikesharp::TYPE_QUATERNION:
+        {
+            script_context.SetResult(new Quaternion(keyValues->GetQuaternion(key)));
+            break;
+        }
+
+        case counterstrikesharp::TYPE_QANGLE:
+        {
+            script_context.SetResult(new QAngle(keyValues->GetQAngle(key)));
+            break;
+        }
+
+        case counterstrikesharp::TYPE_MATRIX3X4:
+        {
+            script_context.SetResult(new matrix3x4_t(keyValues->GetMatrix3x4(key)));
```

Comment:
These should not leak with #613 because the wrapper classes will be `DisposableMemory` and when its no longer used on the managed side these pointers are released